### ISSUE TITLE
A container is destroyed only because it generated a large response

### ISSUE THREAD

**Issue by mcdan** [apache/openwhisk#4750]
If a container sometimes produces replies that are greater than the allowed max size, that container cannot be reused. However, treating this in a similar fashion to unhandled exception or other developer error seems too heavy handed. Layer concurrency on this and you get a lot of failed activations and container churn due to a only slightly misbehaving action.

## Environment details:

* All

## Steps to reproduce the issue:

1. Create an action that produces a response greater than 1mb   
2. Run the action 
2. The container is destroyed
3. Create an action that produces an handled exception
3. The container is destroyed 

## Provide the expected results and outputs:

```
{                                                                                                                                                                       
    "error": "The action produced a response that exceeded the allowed length: 1061670 > 1048576 bytes. The truncated response was:
```
But the container is not destroyed.

## Provide the actual results and outputs:

```
{                                                                                                                                                                       
    "error": "The action produced a response that exceeded the allowed length: 1061670 > 1048576 bytes. The truncated response was:
```
The container is destroyed.

## Additional information you deem important:
* This always happens and can negatively affect the system when running at scale.

**Comment by chetanmeh** [comment]
+1 to not treat it in same way we do with any Developer Error and let the container still run.

**Comment by jiangpengcheng** [comment]
and if action's code return other structure rathen than an object, it will also destroy a container, like

### error.js
```nodejs
function main(params) {
  return ["jack", "tom"];
}
```

**Comment by mcdan** [comment]
This should be fixed now via: #4754

**Comment by rabbah** [comment]
@jiangpengcheng we should open a separate issue to widen the class of such errors even more.
